modules/apps/headless/headless-admin-workflow/headless-admin-workflow-impl/src/main/java/com/liferay/headless/admin/workflow/internal/resource/v1_0/WorkflowDefinitionLinkResourceImpl.java

```diff
@@ -133,24 +137,98 @@ public WorkflowDefinitionLink postWorkflowDefinitionWorkflowDefinitionLink(
 				workflowDefinition.getName(), workflowDefinition.getVersion()));
 	}
 
+	@Override
+	public WorkflowDefinitionLink
+			putWorkflowDefinitionLinkByExternalReferenceCode(
+				String externalReferenceCode,
+				WorkflowDefinitionLink workflowDefinitionLink)
+		throws Exception {
+
+		long groupId = workflowDefinitionLink.getGroupId();
+
+		if (groupId == 0) {
+			groupId = 0L;
+		}
+		else {
+			Group group = _groupService.fetchGroupByExternalReferenceCode(
```

Comment:
Hi @JhossephG , since we are throwing the `noSuchGroupException` we can use the `getGroupByExternalReferenceCode` to avoid the check of `null`.